### PR TITLE
We never trigger component: Remove those events

### PR DESCRIPTION
**Pull request by bratseth**
@jonmv or @mpolden please review

### CODE REVIEW

**Comment by mpolden** [comment]
Not sure this is 100% true. I think FailureRedeployer can still end up triggering component if it's considered dead.

I thought about throwing (or logging) if something triggers component a few weeks ago. Maybe we should do that now.

**Comment by bratseth** [comment]
Yes, I think this could happen and that this fixes it:
Retried jobs isRunning for too long, and:

    public boolean isRunning(Instant timeoutLimit) {
        if ( ! lastTriggered.isPresent()) return false;

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/JobStatus.java [hunk 1]
```diff
@@ -43,7 +43,9 @@ public JobStatus(DeploymentJobs.JobType type, Optional<DeploymentJobs.JobError>
 
         this.type = type;
         this.jobError = jobError;
-        this.lastTriggered = lastTriggered;
+        
+        // Nwever say we triggered component because we don't:
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/JobStatus.java]: "Nwever" -> "Never"

**Comment by mpolden** [comment]
Right. LGTM then (except the typo) :-)

**Pull request merged**